plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,408 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package opensearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	_ "embed"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/olivere/elastic"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Opensearch struct {
+	AuthBearerToken     string          `toml:"auth_bearer_token"`
+	DefaultPipeline     string          `toml:"default_pipeline"`
+	DefaultTagValue     string          `toml:"default_tag_value"`
+	EnableGzip          bool            `toml:"enable_gzip"`
+	EnableSniffer       bool            `toml:"enable_sniffer"`
+	FloatHandling       string          `toml:"float_handling"`
+	FloatReplacement    float64         `toml:"float_replacement_value"`
+	ForceDocumentID     bool            `toml:"force_document_id"`
+	HealthCheckInterval config.Duration `toml:"health_check_interval"`
+	HealthCheckTimeout  config.Duration `toml:"health_check_timeout"`
+	IndexName           string          `toml:"index_name"`
+	ManageTemplate      bool            `toml:"manage_template"`
+	OverwriteTemplate   bool            `toml:"overwrite_template"`
+	Password            string          `toml:"password"`
+	TemplateName        string          `toml:"template_name"`
+	Timeout             config.Duration `toml:"timeout"`
+	URLs                []string        `toml:"urls"`
+	UsePipeline         string          `toml:"use_pipeline"`
+	Username            string          `toml:"username"`
+	Log                 telegraf.Logger `toml:"-"`
+	version             string
+	pipelineName        string
+	pipelineTagKeys     []string
+	tagKeys             []string
+	tls.ClientConfig
+
+	Client *elastic.Client
+}
+
+//go:embed telegrafTemplate.json
+var telegrafTemplate string
+
+type templatePart struct {
+	TemplatePattern string
+	Version         string
+}
+
+func (*Opensearch) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *Opensearch) Connect() error {
+	err := a.initClient()
+	if err != nil {
+		return err
+	}
+
+	if a.FloatHandling == "" {
+		a.FloatHandling = "none"
+	}
```

Comment:
This is also something for the `Init` method.

Reply:
moved to Init